colcon_core/location.py

```diff
@@ -64,23 +65,26 @@ def get_log_path():
     :returns: The base path for logging or None if logging is disabled
     :rtype: Path or None
     """
+    global _log_base_path
     global _log_base_path_env_var
     path = None
-    if _log_base_path_env_var is not None:
-        path = os.environ.get(_log_base_path_env_var)
-        if path == os.devnull:
-            return None
-        if path:
-            path = Path(str(path))
-    global _log_base_path
-    if not path:
-        assert _log_base_path is not None
+    if _log_base_path is not None:
         path = _log_base_path
-    path /= _log_subdirectory
-    return path
+    elif _log_base_path_env_var is not None:
+        path = os.environ.get(_log_base_path_env_var)
+    else:
+        global _log_base_path_default
+        path = _log_base_path_default
+
+    if path == os.devnull:
+        return None
+
+    return Path(str(path)) / _log_subdirectory
```

Comment:
`args.log_base` ends up in `_log_base_path` but the logic above only uses that value if it is not `None` (see line 71). Otherwise it will use `_log_base_path_default` which is set to `log`.

So I don't see how you can end up in this line with `path` being `None`. Can you provide reproducible steps / environment?

Reply:
Actually the `elif` condition checking the env var was wrong. Fixed in a71ad30e58926989d3edd2742def16b2afdd9d47.

Reply:
I'm just running `colcon list` with no special arguments.

It's the second case that's the problem `_log_base_path_env_var = 'COLCON_LOG_PATH'`, not None:
```
elif _log_base_path_env_var is not None:
        path = os.environ.get(_log_base_path_env_var)
```

```
/home/dan/colcon_ws/venv/bin/python -m colcon --log-level debug list
DEBUG:colcon:Parsed command line arguments: Namespace(base_paths='.', build_base='build', ignore_user_meta=False, log_base=None, log_level=10, main=<bound method ListVerb.main of <colcon_package_information.verb.list.ListVerb object at 0x7f26d8fc4438>>, metas='./colcon.meta', mixin=None, mixin_files=None, mixin_verb=('list',), names_only=False, packages_above=None, packages_above_depth=None, packages_end=None, packages_ignore=None, packages_ignore_regex=None, packages_select=None, packages_select_build_failed=False, packages_select_by_dep=None, packages_select_regex=None, packages_select_test_failures=False, packages_skip=None, packages_skip_build_finished=False, packages_skip_by_dep=None, packages_skip_regex=None, packages_skip_test_passed=False, packages_skip_up_to=None, packages_start=None, packages_up_to=None, paths=None, paths_only=False, topological_order=False, verb_extension=<colcon_package_information.verb.list.ListVerb object at 0x7f26d8fc4438>, verb_name='list', verb_parser=<colcon_mixin.mixin.mixin_argument.MixinArgumentDecorator object at 0x7f26d8fd26a0>)
INFO:colcon.colcon_core.location:Using log path 'None/list_2019-11-04_14-40-55'
[0.426s] INFO:colcon.colcon_core.package_discovery:Crawling recursively for packages in '/home/dan/ros2_ws/src'
[0.440s] DEBUG:colcon.colcon_core.package_identification:Package '/home/dan/ros2_ws/src/angles/angles' with type 'ros.ament_cmake' and name 'angles'
...
```

Reply:
Please try again with the already committed fix.